nvidia/gpu.go

```diff
@@ -81,10 +89,27 @@ func (g Utilization) run(cmd *exec.Cmd, gpuCount int, query string, action Actio
 		if err == io.EOF {
 			break
 		}
+
+		linkName, ok := links[gpuIndex]
+		if !ok {
+			linkName = ""
+		}
+
+		var containerName string
+
+		if linkName == "" {
+			containerName = ""
+		} else {
+			containerName = linkName[:strings.LastIndex(linkName, "-")]
+		}
+
+		containerId := GetContainerId(containerName)
 		headers := strings.Split(query, ",")
 		event := common.MapStr{
-			"gpuIndex": gpuIndex,
-			"type":     "nvidiagpubeat",
+			"linkName":    linkName,
```

Comment:
Should we be storing the container name rather than the link name? How will linkName be used by the querier/consumer of the metric?

Reply:
Storing the linkName preserves the assignment of a given GPU to the GPU instance of the container. This information might be useful for containers with multiple GPUs. 
e.g. Container at t1 had physical GPU3 assigned to its GPU0 device and physical GPU0 to its GPU1 device.
Then after a reboot at t2 the GPUs are re-assigned. e.g phys. GPU2 to GPU1 device and phys. GPU1 to PGU1 device.
The ES query using the linkName would look like this:
Select for a specific GPU device: e.g. container GPU dev identified by the linkName:
 "query_string": {
            "query": "linkName:(bdlocal-1-0)"
   }

Select for all GPU devices:
"query_string": {
            "query": "linkName:(bdlocal-1*)",
            "analyze_wildcard": true
          }